cpp/box/Box.h

```diff
@@ -381,6 +382,53 @@ class Box
         });
     }
 
+    //! Calculate distance between two points using boundary conditions
+    /*! \param p_i Particle position
+        \param p_j Particle position to calculate distances from
+    */
+    float computeDistance(const vec3<float>& p_i, const vec3<float>& p_j) const
+    {
+            vec3<float> v_ij = wrap(p_j - p_i);
+            return std::sqrt(dot(v_ij, v_ij));
+    }
+
+    //! Calculate distance between a set of points and query points using boundary conditions
+    /*! \param points Particle positions
+        \param query_points Particle position to calculate distances from
+        \param dist Distances between points and query_points
+        \param Nvecs The number of points to calculate distances between
+    */
+    void computeDistances(vec3<float>* points, vec3<float>* query_points, float *dist, unsigned int Nvecs) const
+    {
+        util::forLoopWrapper(0, Nvecs, [=](size_t begin, size_t end) {
+            for (size_t i = begin; i < end; ++i)
+            {
+                dist[i] = computeDistance(points[i], query_points[i]);
+            }
+        });
+    }
+
+    //! Calculate distance between a set of points and query points using boundary conditions
+    /*! \param points Particle positions
+        \param query_points Particle position to calculate distances from
+    */
+    void computeAllDistances(vec3<float>* points, vec3<float>* query_points,
+        float* dist, unsigned int Nvecs, unsigned int Mvecs) const
+    {
+        util::forLoopWrapper(0, Nvecs, [=](size_t begin_n, size_t end_n) {
+            for (size_t i = begin_n; i < end_n; ++i)
+            {
+                util::forLoopWrapper(0, Mvecs, [=](size_t begin_m, size_t end_m) {
+                    for (size_t j = begin_m; j < end_m; ++j)
+                    {
+                        dist[i*Mvecs + j] = computeDistance(points[i], query_points[j]);
+                    }
+                });
+            }
+        std::cout.flush();
```

Comment:
You'll want to remove this line, which I assume was for testing. Also remove the `iostream` above.